File: DataStructure/2.QueueAndDeque/Group2/hyunjin/18258.py
import sys

class Queue:
    def __init__(self):
        self._items = []
        self.first_idx = 0

    def push(self, X):
        self._items.append(X)

    def empty(self):
        if len(self._items) == self.first_idx:
            self._items = []
            self.first_idx = 0
            return 1
        return 0

    def pop(self):
        if len(self._items) > self.first_idx:
            self.first_idx += 1
            return self._items[self.first_idx - 1]
        return -1

    def size(self):
        return len(self._items) - self.first_idx

    def front(self):
        if len(self._items) > self.first_idx:
            return self._items[self.first_idx]
        return -1

    def back(self):
        if len(self._items) > self.first_idx:
            return self._items[-1]
        return -1

queue = Queue()
N = int(sys.stdin.readline())

for _ in range(N):
    order = sys.stdin.readline().rstrip()
    if "push" in order:
        __, num = order.split()
        queue.push(int(num))
    elif order == "front":
        print(queue.front())
    elif order == "back":
        print(queue.back())
    elif order == "pop":
        print(queue.pop())
    elif order == "size":
        print(queue.size())
    else:
        print(queue.empty())

File: DataStructure/2.QueueAndDeque/Group1/seoyoon/18258.cpp
#include <iostream>
#include <string>
#include <queue>
using namespace std;

int main()
{
	//iostream과 stdio동기화 비활성화 -> 속도 향상
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);

	int num, ch;
	string s;

	cin >> num;

	queue<int> que;

	for (int i = 0; i < num; i++)
	{
		cin >> s;

		if (s == "push")
		{
			cin >> ch;
			que.push(ch);
		}
		else if (s == "pop")
		{
			if (que.empty())
				cout << -1 << '\n';
			else
			{
				cout << que.front() << '\n';
				que.pop();
			}
		}
		else if (s == "front")
		{
			cout << (que.empty() ? -1 : que.front()) << '\n';
		}
		else if (s == "back")
		{
			cout << (que.empty() ? -1 : que.back()) << '\n';
		}
		else if (s == "size")
		{
			cout << que.size() << '\n';
		}
		else if (s == "empty")
		{
			cout << que.empty() << '\n';
		}
	}
	return 0;
}